lib/coinbase/smart_contract.rb

```diff
@@ -126,6 +126,72 @@ def self.create_multi_token_contract(
       new(contract)
     end
 
+    def self.read(
+      network_id:,
```

Comment:
We should take a `network:` that could be either a `Coinbase::Network` or an ID:

https://github.com/coinbase/coinbase-sdk-ruby/blob/4352c77e3ab4c66ea5bcf69e0dbd1bf7aadd7594/lib/coinbase/transfer.rb#L39